app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+package com.rockthevote.grommet.ui.registration
+
+import android.util.Base64
+import com.rockthevote.grommet.R
+import com.rockthevote.grommet.data.db.model.*
+import com.rockthevote.grommet.ui.registration.address.AddressData
+import com.rockthevote.grommet.ui.registration.name.PersonNameData
+import com.rockthevote.grommet.util.Dates
+import java.util.*
+
+/**
+ * Handles transformation of [registrationData]
+ */
+class RegistrationDataTransformer @Throws(InvalidRegistrationException::class) constructor(
+    private val registrationData: RegistrationData,
+    private val partnerInformation: PartnerInformation
+) {
+
+    init {
+        validateRegistrationData(registrationData)
+    }
+
+    // These have already be evaluated as non null in [validateRegistrationData]
+    private val newRegistrationData = registrationData.newRegistrantData!!
+    private val addressData = registrationData.addressData!!
+    private val additionalInfoData = registrationData.additionalInfoData!!
+    private val assistanceData = registrationData.assistanceData!!
+    private val reviewData = registrationData.reviewData!!
+
+    /**
+     * Transforms the [registrationData] and [partnerInformation] into a [RockyRequest]
+     *
+     * @throws [InvalidRegistrationException] if unable to validate [RegistrationData]
+     */
+    fun transform(
+        unknown: UnknownDataSource
+    ): RockyRequest {
+
+        val body = RockyRequestBody(
+            lang = reviewData.formLanguage,
+            phoneType = null, // TODO I can't find where/what is supposed to populate this
```

Comment:
I think this is misplaced, see previous comment on phonetype

Reply:
I think I'm understanding this a bit better now then? The type in this case was initially there, but served no purpose so it was removed?